lib/network_monitor.c

```diff
@@ -119,52 +157,80 @@ mptcpd_sockaddr_create(struct mptcpd_rtm_addr const *info)
         if (family != AF_INET && family != AF_INET6)
                 return NULL;
 
-        struct sockaddr *addr = NULL;
+        struct mptcpd_addr_info *const ai =
+                        l_new(struct mptcpd_addr_info, 1);
+        ai->address.ss_family = family;
+        ai->count = 1;
 
         if (family == AF_INET) {
-                struct sockaddr_in *const a =
-                        l_new(struct sockaddr_in, 1);
-
-                a->sin_family = family;
-
+                struct sockaddr_in *const a = (struct sockaddr_in *const) &ai->address;
```

Comment:
I don't think you need the `const` in the cast in this case since only the pointer `a` is `const`, not the underlying `struct sockaddr_in` type.
```suggestion
                struct sockaddr_in *const a = (struct sockaddr_in *) &ai->address;
```